finesse/gui/measure_script/parse.py

```diff
@@ -0,0 +1,120 @@
+"""Code for parsing the YAML-formatted measure scripts."""
+import logging
+from dataclasses import dataclass
+from io import TextIOBase
+from pathlib import Path
+from typing import Any, Dict, Optional, Sequence, Union
+
+import yaml
+from pubsub import pub
+from PySide6.QtWidgets import QWidget
+from schema import And, Or, Schema, SchemaError
+
+from ...config import ANGLE_PRESETS
+from ..error_message import show_error_message
+
+
+@dataclass
+class Measurement:
+    """Represents a single measurement (i.e. angle + number of measurements)."""
```

Comment:
Should this be called `Step` rather than `Measurement` to avoid mixing it with the individual acquisitions?

Reply:
Good point.